library/alloc/src/vec/mod.rs

```diff
@@ -3025,26 +3025,29 @@ impl<T: Clone, A: Allocator> Vec<T, A> {
         self.spec_extend(other.iter())
     }
 
-    /// Copies elements from `src` range to the end of the vector.
+    /// Copies elements from one part of the vector to the end.
     ///
-    /// # Panics
+    /// `src` is the range that the elements will be copied from.
     ///
-    /// Panics if the starting point is greater than the end point or if
-    /// the end point is greater than the length of the vector.
+    /// # Panics if:
+    ///
+    /// - The starting index is greater than the end index.
+    /// - The end index is greater than the length of the vector.
     ///
     /// # Examples
     ///
     /// ```
-    /// let mut vec = vec![0, 1, 2, 3, 4];
-    ///
-    /// vec.extend_from_within(2..);
-    /// assert_eq!(vec, [0, 1, 2, 3, 4, 2, 3, 4]);
+    /// let mut characters1 = vec!['a', 'b', 'c', 'd', 'e'];
+    /// characters1.extend_from_within(2..);
+    /// assert_eq!(characters1, ['a', 'b', 'c', 'd', 'e', 'c', 'd', 'e']);
     ///
-    /// vec.extend_from_within(..2);
-    /// assert_eq!(vec, [0, 1, 2, 3, 4, 2, 3, 4, 0, 1]);
+    /// let mut characters2 = vec!['a', 'b', 'c', 'd', 'e'];
+    /// characters2.extend_from_within(..2);
+    /// assert_eq!(characters2, ['a', 'b', 'c', 'd', 'e', 'a', 'b']);
     ///
-    /// vec.extend_from_within(4..8);
-    /// assert_eq!(vec, [0, 1, 2, 3, 4, 2, 3, 4, 0, 1, 4, 2, 3, 4]);
+    /// let mut characters3 = vec!['a', 'b', 'c', 'd', 'e'];
+    /// characters3.extend_from_within(1..3);
+    /// assert_eq!(characters3, ['a', 'b', 'c', 'd', 'e', 'b', 'c']);
```

Comment:
I agree with #104762, however I'd probably make the examples a little bit more varied and include some Non-`Copy` but `Clone` types. Preferably I'd like to see `char`s, `i32`, and `String` as examples.